src/main/java/uk/gov/hmcts/dts/fact/services/admin/list/AdminAreasOfLawService.java

```diff
@@ -19,10 +23,63 @@ public AdminAreasOfLawService(final AreasOfLawRepository areasOfLawRepository) {
         this.areasOfLawRepository = areasOfLawRepository;
     }
 
+    public AreaOfLaw getAreaOfLaw(final Integer id) {
+        try {
+            return new AreaOfLaw(areasOfLawRepository.getOne(id));
+        } catch (final javax.persistence.EntityNotFoundException exception) {
+            throw new NotFoundException(exception);
+        }
+    }
+
     public List<AreaOfLaw> getAllAreasOfLaw() {
         return areasOfLawRepository.findAll()
             .stream()
             .map(AreaOfLaw::new)
             .collect(toList());
     }
+
+    @Transactional
+    public AreaOfLaw updateAreaOfLaw(final AreaOfLaw updatedAreaOfLaw) {
+        // Ensure entity with ID exists
+        final Optional<uk.gov.hmcts.dts.fact.entity.AreaOfLaw> areaOfLawEntity =
+            areasOfLawRepository.findById(updatedAreaOfLaw.getId());
+        if (areaOfLawEntity.isEmpty()) {
+            throw new NotFoundException(updatedAreaOfLaw.getId().toString());
+        }
+
```

Comment:
Nothing wrong with the previous lines, but I think it is nicer like below without the need to use 'optional':
AreaOfLaw areaOfLawEntity = areasOfLawRepository.findById(updatedAreaOfLaw.getId())
            .orElseThrow(() -> new NotFoundException(updatedAreaOfLaw.getId().toString()));